### PR TITLE
Add support for Arduino Outdoor Carrier

### PR DESCRIPTION
**Pull request by manchoz**
Add support for SPIFFS/W25Q16DV on Arduino Outdoor Carrier.

### CODE REVIEW

File: src/Arduino_SPIFFS.h [hunk 1]
```diff
@@ -59,14 +59,14 @@ static_assert(sizeof(u32_t) == sizeof(unsigned int), "Arduino SPIFFS Wrapper - s
...
-static uint16_t constexpr READ_WRITE = SPIFFS_O_RDWR;
-static uint16_t constexpr DIRECT     = SPIFFS_O_DIRECT; /* Any writes to the filehandle will never be cached but flushed directly */
-static uint16_t constexpr EXCLUSIVE  = SPIFFS_O_EXCL;
+static uint16_t constexpr APPEND        = SPIFFS_O_APPEND;
+static uint16_t constexpr TRUNCATE      = SPIFFS_O_TRUNC;
+static uint16_t constexpr CREATE        = SPIFFS_O_CREAT;
+static uint16_t constexpr READ_ONLY     = SPIFFS_O_RDONLY;
+static uint16_t constexpr WRITE_ONLY    = SPIFFS_O_WRONLY;
+static uint16_t constexpr READ_WRITE    = SPIFFS_O_RDWR;
+static uint16_t constexpr WRITE_DIRECT  = SPIFFS_O_DIRECT; /* Any writes to the filehandle will never be cached but flushed directly */
```
Comment (aentinger) [inline src/Arduino_SPIFFS.h:68]: @manchoz why this rename? Is there a top level header define `DIRECT` in some other code? If so I'd prefer the change to happen over there, as this constitutes a breaking API change.

**Review by aentinger — CHANGES_REQUESTED** [review]
Apart from my comment on the renaming this PR looks good 👍  However, I couldn't find any reference to the Arduino Outdoor Carrier elsewhere? Can you point me towards a schematic via internal channels?

File: src/Arduino_SPIFFS.h (hunk 1 above)
Comment (manchoz) [reply src/Arduino_SPIFFS.h:68]: Yes, here it is [PID_v1.h#L16](https://github.com/arduino-libraries/MKRMotorCarrier/blob/518789127fbaa16ff8810706deb674f7b7b124f7/src/src/PID/PID_v1.h#L16).

Comment (aentinger) [reply src/Arduino_SPIFFS.h:68]: Defines in header files are evil 😢 . How about changing the include order that `PID_v1.h` is always included after `Arduino_MKRMEM`. Also [MKRMotorCarrier](https://github.com/arduino-libraries/MKRMotorCarrier) is archived, is it still relevant?

Comment (manchoz) [reply src/Arduino_SPIFFS.h:68]: [MKRMotorCarrier](https://github.com/arduino-libraries/MKRMotorCarrier) has been replaced by [ArduinoMotorCarrier](https://github.com/arduino-libraries/ArduinoMotorCarrier/) which has the same issue [PID_v1.h#L16](https://github.com/arduino-libraries/ArduinoMotorCarrier/blob/a4c29bddadcfab5a019061a4a1140135188c1e92/src/src/PID/PID_v1.h#L16).

I don't have a position about the include order.

Comment (aentinger) [reply src/Arduino_SPIFFS.h:68]: And would doing something like this be an option:
```C++
#ifdef DIRECT
#undef DIRECT
#endif
static uint16_t constexpr DIRECT  = SPIFFS_O_DIRECT; /* Any writes to the filehandle will never be cached but flushed directly */

```

Comment (manchoz) [reply src/Arduino_SPIFFS.h:68]: Ok. I will made the change.

**Review by aentinger — APPROVED** [review]
LGTM :+1:

**Pull request merged**